src/kimmdy/topology/topology.py

```diff
@@ -1323,6 +1324,7 @@ def bind_bond(
                         "not registed in moleculetype.radicals"
                     )
 
+        # TODO: Do we still need this with grappa?
```

Comment:
I think it makes the resulting topology more correct if we update the residuetype and atomtype, so I'd keep it. not sure why 'quickfix', seems fine to me.

Reply:
It was a bit of a quickfix with a performance hit when we added it, because it would have to iterate over all atoms to find the correct residue, but I have now made that part make use of connectivity information instead.